### DISCUSSION TITLE
Failure deployed on Github Pages

### DISCUSSION THREAD

**Discussion by ItProgrammist** [community/community#23609] [New to GitHub]
<p>Good day! I ran into such a problem, no matter what I try to make changes to my repository on Github Pages, I constantly get “Failure” in Environments, please tell me how can I make changes to my repository on Github Pages without such an error? Help me please!</p>

**Comment by jfullstackdev** [comment]
<p>is this the one you are referring?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/8/c80d4c9dfe5dd91f66853439881ac3b6ed9792fd.jpeg" data-download-href="/uploads/short-url/sxJZeDn3hQPNJ06G7fQnqxfUnU1.jpeg?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181078194-681d333e-be31-497f-a146-32892235599c.jpeg" alt="image" data-base62-sha1="sxJZeDn3hQPNJ06G7fQnqxfUnU1" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/8/c80d4c9dfe5dd91f66853439881ac3b6ed9792fd_2_690x373.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/c/8/c80d4c9dfe5dd91f66853439881ac3b6ed9792fd_2_1035x559.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/8/c80d4c9dfe5dd91f66853439881ac3b6ed9792fd_2_1380x746.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/8/c80d4c9dfe5dd91f66853439881ac3b6ed9792fd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1909×1034 60.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s working correctly, where is the error?</p>

**Comment by ItProgrammist** [comment]
<p>I mean, if I try to add something to the repository on Gihub Pages, some changes to my website, Github Pages doesn’t update them, but writes a build error.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/3/83559f3232faef74a1bbedcc7ecc5218f617e62f.png" data-download-href="/uploads/short-url/iJPZxCu8oYrq0wS3fDFDACMRmft.png?dl=1" title="That&amp;#39;s a problem" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50521864/181078195-4f3e34e9-8e76-4323-afd0-6d21647ba433.png" alt="That's a problem" data-base62-sha1="iJPZxCu8oYrq0wS3fDFDACMRmft" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/3/83559f3232faef74a1bbedcc7ecc5218f617e62f_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/3/83559f3232faef74a1bbedcc7ecc5218f617e62f_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/3/83559f3232faef74a1bbedcc7ecc5218f617e62f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/3/83559f3232faef74a1bbedcc7ecc5218f617e62f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">That&amp;#39;s a problem</span><span class="informations">1366×768 52.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
I tried to add or change something on Github Pages but gave an error every time. All errors are indicated in the photo in the Deployment history. Help please

**Comment by jfullstackdev** [comment]
<p>is that GitHub Action? also, I cannot see the specific error of each</p>
<p>also another thing, there was a post here about the problem of submodules also on GitHub Pages, so he simply removed it, and his GitHub Pages was resolved, I’m not saying you have the same error, but it might be the case</p>

**Comment by ItProgrammist** [comment]
<p>These errors show up on GitHub Pages.  As soon as I make any change to the repository, I get this error for each change (for example, I made changes to index.html and style.css, tried to add files to a new folder (only .html, .css, .js))<br>
I also thought that I needed to delete this repository, and then re-create the same one, but not the fact that it will work again.</p>

**Comment by jfullstackdev** [comment]
<p>I just tested now deploying a site then changing a content, the deployment does not give me error.</p>
<p>so I checked again the errors,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/a/5a1f984f77d85138cb6e90f2850ce48eaf97c3e7.png" data-download-href="/uploads/short-url/cRgFbjVVeALaASBTQeVaTScTYNh.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181078198-15b0c37e-def5-43b7-b5b8-5f47800eb0b1.png" alt="image" data-base62-sha1="cRgFbjVVeALaASBTQeVaTScTYNh" width="690" height="217" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/a/5a1f984f77d85138cb6e90f2850ce48eaf97c3e7_2_690x217.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/a/5a1f984f77d85138cb6e90f2850ce48eaf97c3e7_2_1035x325.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/5/a/5a1f984f77d85138cb6e90f2850ce48eaf97c3e7_2_1380x434.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/a/5a1f984f77d85138cb6e90f2850ce48eaf97c3e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1899×599 35.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>when I click the link, it says</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/5/35ac0d0c76ef231df1d370ae4a9ef8e7a9527536.png" data-download-href="/uploads/short-url/7ENVfHCO7p0qufbGvOXvVwTyek6.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181078199-812156bd-c3cb-489a-874f-1b7405849bfb.png" alt="image" data-base62-sha1="7ENVfHCO7p0qufbGvOXvVwTyek6" width="690" height="48" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/5/35ac0d0c76ef231df1d370ae4a9ef8e7a9527536_2_690x48.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/5/35ac0d0c76ef231df1d370ae4a9ef8e7a9527536_2_1035x72.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/5/35ac0d0c76ef231df1d370ae4a9ef8e7a9527536_2_1380x96.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/5/35ac0d0c76ef231df1d370ae4a9ef8e7a9527536_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1729×122 12.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s unauthorized access that’s why it’s an error</p>
<p><code>"The user 'System:PublicAccess;aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' is not authorized to access this resource."</code></p>

**Comment by ItProgrammist** [comment]
<p>I haven’t looked at the errors in this section yet, they showed up as alerts in the Environments section, as shown in the photo I showed earlier.  And if, I have such an error, as shown in your photo (I have not seen such an error), then in connection with what can there be unauthorized access?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>When debugging, start with the <em>first</em> error. Later errors might be caused by the first failing step. In your case, that’s during checkout in the Pages build job: <a href="https://github.com/ItProgrammist/ItProgrammist.github.io/runs/4680789953?check_suite_focus=true#step:3:203" class="inline-onebox">Trigger rebuild · ItProgrammist/ItProgrammist.github.io@fd40c1e · GitHub</a></p>
<p>It looks like your repository contains a broken submodule configuration, so checkout fails, and because of that the Pages build never happens. How to fix the submodule depends on what the submodule should be. The name of the submodule looks like the repository references itself, which doesn’t really make sense. In that case just deleting the submodule should be enough. :slightly_smiling_face:</p>

**Comment by ItProgrammist** [comment]
<p>Please tell me, how can I delete this submodule?</p>

**Comment by airtower-luna** [comment]
<p>Just delete it (in your local copy of the repository):</p>
<pre><code class="lang-auto">git rm ItProgrammist.github.io
</code></pre>
<p>Then commit and push as usual. The diff should show you that a submodule (or “subproject”) is getting removed.</p>

**Comment by ItProgrammist** [comment]
<p>I already tried to do this, but this error appears:<br>
<img src="https://user-images.githubusercontent.com/50521864/181078204-ac58792a-3746-49a9-a514-70a1b0a9e926.png" alt="Another problem" data-base62-sha1="MwgIirWrXOj2k0BeEaIWdAb7xq" width="562" height="364"></p>

**Comment by airtower-luna** [comment]
<p>Huh, that’s odd. I wonder if your local repository isn’t at the same state as the one on GitHub? You can check <code>git status</code> and <code>ls -al</code> to compare.</p>

**Comment by ItProgrammist** [comment]
<p>I don’t understand what happened at all, but it worked. I just added a file, made a commit, made a push, and it worked. The only thing that confused me when I made the commit was: delete mode 160000 <a href="http://ItProgrammist.github.io" rel="noopener nofollow ugc">ItProgrammist.github.io</a> in the first photo<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/8/98930ee0e619f60e002e2fc334b07f12a1a35f54.png" data-download-href="/uploads/short-url/lLJE3vSVCFIedMHXaAhbUcUBlaY.png?dl=1" title="shot1" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50521864/181078206-a5061c0c-06fd-402b-9864-506325af537b.png" alt="shot1" data-base62-sha1="lLJE3vSVCFIedMHXaAhbUcUBlaY" width="690" height="367" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/8/98930ee0e619f60e002e2fc334b07f12a1a35f54_2_690x367.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/8/98930ee0e619f60e002e2fc334b07f12a1a35f54_2_1035x550.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/8/98930ee0e619f60e002e2fc334b07f12a1a35f54.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/8/98930ee0e619f60e002e2fc334b07f12a1a35f54_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">shot1</span><span class="informations">1366×727 44.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ItProgrammist** [comment]
<p>Here another problem arose, I don’t know in connection with what, but I’m interested. All files that are in the folders in the subPages folder have exactly the same errors.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/e/4e79b1cdf5d956cc854df1492b490496d72dd823.png" data-download-href="/uploads/short-url/bcdYgBZJlkVRopIdW3FEiZbaydR.png?dl=1" title="new one" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50521864/181078207-655e9e56-43d7-4af2-b8e7-9be658629c62.png" alt="new one" data-base62-sha1="bcdYgBZJlkVRopIdW3FEiZbaydR" width="690" height="330" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e79b1cdf5d956cc854df1492b490496d72dd823_2_690x330.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e79b1cdf5d956cc854df1492b490496d72dd823_2_1035x495.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/e/4e79b1cdf5d956cc854df1492b490496d72dd823.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e79b1cdf5d956cc854df1492b490496d72dd823_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">new one</span><span class="informations">1366×655 222 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="ItProgrammist" data-post="13" data-topic="220671">
<div class="title">
<div class="quote-controls"></div>
 ItProgrammist:</div>
<blockquote>
<p>The only thing that confused me when I made the commit was: delete mode 160000 <a href="http://ItProgrammist.github.io">ItProgrammist.github.io</a> in the first photo</p>
</blockquote>
</aside>
<p>That means the deletion of the submodule was committed there. The mode describe access permissions, but I’m not sure how those work on Windows. :wink:</p>

**Comment by airtower-luna** [comment]
<p>That’s an error in your code, and the error message says exactly what the problem is. Load your scripts over HTTPS, not HTTP.</p>

**Comment by ItProgrammist** [comment]
<p>But before that they worked correctly, I did not change anything. Now I will look at my code again</p>

**Comment by airtower-luna** [comment]
<p>Depends on when you last checked the site. Refusing to load scripts over HTTP on HTTPS sites is a security feature browsers gradually introduced over the last few years.</p>

**Comment by ItProgrammist** [comment]
<p>I had two same errors, I fixed everything from http to https. One’s gone, the other stayed<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/1/61c4d901b0680a81e7e34710a82e604fe489a163.png" data-download-href="/uploads/short-url/dWU3EcWrn5EUNdSRnQcPM1gNCoz.png?dl=1" title="This one" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50521864/181078209-58b179ca-d3a6-4823-9fec-ff138b3b597c.png" alt="This one" data-base62-sha1="dWU3EcWrn5EUNdSRnQcPM1gNCoz" width="690" height="329" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/1/61c4d901b0680a81e7e34710a82e604fe489a163_2_690x329.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/1/61c4d901b0680a81e7e34710a82e604fe489a163_2_1035x493.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/6/1/61c4d901b0680a81e7e34710a82e604fe489a163.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/1/61c4d901b0680a81e7e34710a82e604fe489a163_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">This one</span><span class="informations">1366×653 173 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>That’s essentially the same thing, just for a font instead of a script.</p>
